crates/y-sweet-core/src/store/s3.rs

```diff
@@ -1,17 +1,27 @@
-use crate::config::S3Config;
+use crate::store::Store;
 use anyhow::Result;
 use async_trait::async_trait;
-use reqwest::{Client, StatusCode};
+use reqwest::{Client, StatusCode, Url};
 use rusty_s3::{Bucket, Credentials, S3Action};
-use std::{cell::RefCell, time::Duration};
+use serde::{Deserialize, Serialize};
+use std::sync::RwLock;
+use std::time::Duration;
 use time::OffsetDateTime;
-use url::Url;
-use y_sweet_core::store::Store;
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct S3Config {
+    pub key: String,
+    pub endpoint: String,
+    pub secret: String,
+    pub bucket: String,
+    pub region: String,
+    pub bucket_prefix: Option<String>,
+}
 
 const PRESIGNED_URL_DURATION_SECONDS: u64 = 60 * 60;
 pub struct S3Store {
     bucket: Bucket,
-    _bucket_inited: RefCell<bool>,
+    _bucket_inited: RwLock<bool>,
```

Comment:
can this just be an `AtomicBool`?

Reply:
yep, I've use SeqCst ordering cause errors should be more easily reproducible.